smslibrary/src/test/java/com/eis/smslibrary/SMSMessageHandlerTest.java

```diff
@@ -0,0 +1,46 @@
+package com.eis.smslibrary;
+
+import com.eis.communication.MessageParseStrategy;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Luca Crema
+ */
+public class SMSMessageHandlerTest {
+
+    SMSMessageHandler defaultInstance;
+    private static final MessageParseStrategy<String,SMSPeer,SMSMessage> DEFAULT_SMS_MESSAGE_PARSE_STRATEGY = SMSMessageHandler.getInstance().new DefaultSMSMessageParseStrategy();
+    private static final SMSPeer DEFAULT_PEER = new SMSPeer("+393465433432");
+    private static final String DEFAULT_MESSAGE_CONTENT = "Test content";
+    private static final SMSMessage DEFAULT_MESSAGE = new SMSMessage(DEFAULT_PEER,DEFAULT_MESSAGE_CONTENT);
+
+    @Before
+    public void init(){
+        defaultInstance = SMSMessageHandler.getInstance();
+
+        defaultInstance.setMessageParseStrategy(DEFAULT_SMS_MESSAGE_PARSE_STRATEGY);
+    }
+
+    @Test
+    public void getInstance_isNotNull() {
+        Assert.assertNotNull(SMSMessageHandler.getInstance());
+    }
+
+
+    @Test
+    public void parseMessage_message_isCorrect() {
+        Assert.assertEquals(DEFAULT_MESSAGE.getData(),defaultInstance.parseMessage(DEFAULT_PEER.getAddress(), defaultInstance.parseData(DEFAULT_MESSAGE)).getData());
+    }
+
+    /**
+     * If a normal sms is parsed (without hidden character) it should return null
+     */
+    @Test
+    public void parseMessage_wrongSMS_isNull(){
```

Comment:
_parseMessage_smsData_hasNotHiddenChar()_ it's better, no?

Reply:
Actually the naming convention for test methods I use has this structure:
- Method name
- Given input
- Expected output
So I could go for something like `parseMessage_messageWithoutHiddenChar_isNull()`, but is this actually more descriptive?

Reply:
Well, now that i know your name convention: if you want you can write as you said: "parseMessage_messageWithoutHiddenChar_isNull()" and remove the specification.
Or you can leave it as is.

Reply:
I'm leaving it as it is, don't think it's worth edit